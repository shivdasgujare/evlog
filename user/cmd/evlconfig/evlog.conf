#
# Configuration values for evlogd daemon. 
# Use 'evlconfig' command to update these values. If this file is edited 
# manually, the updated value will not be used by the evlogd daemon. So,
# the evlogd daemon has to be restarted. Hence, use 'evlconfig' command
# to send this values to evlogd daemon.
# Console display level - OFF=111
#
Discard Dups: on
Discard Interval: 3
Discard Count: 100
Event Screen: nofilter
Console display level: 111
Lookback Size: 3
