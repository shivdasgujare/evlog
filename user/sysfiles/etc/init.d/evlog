#!/bin/sh
#
# chkconfig: 35 09 90
# description: Enterprise event log daemon
#
#

# Source function library.

REDHAT=`cat /proc/version | grep "Red Hat"`
# Source function library.
if [ ! -z "$REDHAT" ]
then
	. /etc/init.d/functions
fi


[ -f /sbin/evlogd ] || exit 0

EVLOGD_OPTIONS=""
# Source config
if [ -f /etc/sysconfig/evlog ]; then
	. /etc/sysconfig/evlog
fi

RETVAL=0

umask 077

record_last_recid() {
	/sbin/evlview -l $1 -t 1 -S %recid% > $2 2> /dev/null
	test -s $2 || echo 0 > $2
	chmod 644 $2
}

start() {
 	echo -n $"Starting enterprise event logger: "

	record_last_recid /var/evlog/eventlog /var/evlog/bootrecid
	record_last_recid /var/evlog/privatelog /var/evlog/pbootrecid
 	
 	if [ ! -z "$REDHAT" ]
	then 	
		daemon /sbin/evlogd $EVLOGD_OPTIONS
	else
		/sbin/evlogd $EVLOGD_OPTIONS
	fi
	sleep 1
	RETVAL=$?
	echo
	touch /var/lock/subsys/evlog
	return $RETVAL
}	
stop() {
	echo -n $"Shutting down enterprise event logger: "
	
	if [ ! -z "$REDHAT" ]
	then
		killproc evlogd
	else
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		if [ ! -z "$PID" ]
		then
			kill -s SIGTERM $PID
		fi
	fi

	RETVAL=$?
	echo
	rm -f /var/lock/subsys/evlog
	if [ -e /var/run/evlogd.pid ]
	then
		rm -f /var/run/evlogd.pid
	fi
	return $RETVAL
}
rhstatus() {
	status evlogd
}
restart() {
	stop
	start
}	

case "$1" in
  start)
  	start
	;;
  stop)
  	stop
	;;
  status)
  	rhstatus
	;;
  restart|reload)
  	restart
	;;
  condrestart)
    restart	
	;;
  *)
	echo $"Usage: $0 {start|stop|status|restart|condrestart}"
	exit 1
esac

exit $?

