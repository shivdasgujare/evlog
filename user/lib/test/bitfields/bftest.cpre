#include <stdlib.h>
#include <stdio.h>
#include <errno.h>
#include <wchar.h>

#include <posix_evlog.h>
#include <evlog.h>

#define ARRAY_FMT
#define STRARRAY_FMT

typedef unsigned char uchar;
typedef signed char schar;
typedef unsigned long long ulonglong;
typedef long long longlong;
typedef wchar_t wchar;

typedef int xint;
typedef uint uxint;

enum e { sun, mon, tue, wed, thu, fri, sat };
struct b1 {
	char c;
};
struct b2 {
	char c1:4;
	char c2:4;
};

int exitStatus = 0;

static void
checkStatus(int status, int evty)
{
	if (status != 0) {
		fprintf(stderr, "write of event type %d failed:\n", evty);
		errno = status;
		perror("posix_log_write:");
		exitStatus = 1;
	}
}

main()
{
	posix_log_facility_t facility = 8;
	int event_type;
	posix_log_severity_t severity = LOG_DEBUG;
	int status;

