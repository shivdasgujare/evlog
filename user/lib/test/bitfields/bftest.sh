#!/bin/bash
#
TEST_TEMPLATE=bftest.out

EVLTMPLPATH=$EVL_TESTDIR/templates
export EVLTMPLPATH

if [ "$USER" = root ]
then
	/sbin/evltc -c $EVLTMPLPATH/user/bftest.t
else
echo >&2 "bftest.sh: Not running as root.  Assuming template source file"
echo >&2 "$EVLTMPLPATH/user/bftest.t has already been compiled."
fi

logtime=`date +%s`
sleep 1
./bftest
sleep 1
/sbin/evlview -f "(time>${logtime} && event_type>3000 && event_type<3200)" -S '%event_type:d%: %data%' > bftest.sh.out

diff bftest.sh.out $TEST_TEMPLATE

if [ $? -eq "0" ]
then
	echo templates bitfield test	:PASSED
else
	echo templates bitfield test	:FAILED
fi
