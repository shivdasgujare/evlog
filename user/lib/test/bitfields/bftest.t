#define ARRAY_FMT "(%d,)"
#define STRARRAY_FMT "(%Z,)"
typedef int xint;
typedef uint uxint;

struct b1;
aligned attributes {
	char c;
}
format string "{%c%}"
END

struct b2;
aligned attributes {
	char c1:4;
	char c2:4;
}
format string "{%c1%,%c2%}"
END
facility 8; event_type 13001;
aligned attributes {
	int	bf1_i:2;
	short	bf1_s:2;
	char	bf1_c:2;
}
format string "natts = 3 \t%bf1_i% %bf1_s% %bf1_c% "
END
facility 8; event_type 13002;
aligned attributes {
	char	bf1_c:2;
	short	bf1_s:2;
	int	bf1_i:2;
}
format string "natts = 3 \t%bf1_c% %bf1_s% %bf1_i% "
END
facility 8; event_type 13003;
aligned attributes {
	char	bf1_c:2;
	char	bf1_c2:2;
}
format string "natts = 2 \t%bf1_c% %bf1_c2% "
END
facility 8; event_type 13004;
aligned attributes {
	char	bf1_c:2;
	char	bf1_c2:2;
	char	bf1_c3:2;
	char	bf1_c4:2;
	char	bf1_c5:2;
}
format string "natts = 5 \t%bf1_c% %bf1_c2% %bf1_c3% %bf1_c4% %bf1_c5% "
END
facility 8; event_type 13005;
aligned attributes {
	int	bf1_i:2;
	int	bf1_i2:2;
	int	bf1_i3:2;
	int	bf1_i4:2;
	int	bf1_i5:2;
}
format string "natts = 5 \t%bf1_i% %bf1_i2% %bf1_i3% %bf1_i4% %bf1_i5% "
END
facility 8; event_type 13006;
aligned attributes {
	short	bf1_s:2;
	char	bf1_c2:2;
	char	bf1_c3:2;
	char	bf1_c4:2;
}
format string "natts = 4 \t%bf1_s% %bf1_c2% %bf1_c3% %bf1_c4% "
END
facility 8; event_type 13007;
aligned attributes {
	char	bf1_c2:2;
	char	bf1_c3:2;
	char	bf1_c4:2;
	short	bf1_s:2;
}
format string "natts = 4 \t%bf1_c2% %bf1_c3% %bf1_c4% %bf1_s% "
END
facility 8; event_type 13008;
aligned attributes {
	char	bf1_c:7;
	short	bf1_s:8;
	int	bf1_i:16;
}
format string "natts = 3 \t%bf1_c% %bf1_s% %bf1_i% "
END
facility 8; event_type 13010;
aligned attributes {
	char	bf1_c1:3;
	char	bf1_c2:3;
	char	bf1_c3:3;
	char	bf1_c4:3;
	char	bf1_c5:3;
}
format string "natts = 5 \t%bf1_c1% %bf1_c2% %bf1_c3% %bf1_c4% %bf1_c5% "
END
facility 8; event_type 13011;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13012;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	char	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13013;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	int	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13014;
aligned attributes {
	short	bf1_s1:3;
	char	:0;
	short	bf1_s2:3;
	short	bf1_s3:3;
}
format string "natts = 3 \t%bf1_s1% %bf1_s2% %bf1_s3% "
END
facility 8; event_type 13015;
aligned attributes {
	char	bf1_c1:3;
	short	:0;
	char	bf1_c2:3;
	char	bf1_c3:3;
}
format string "natts = 3 \t%bf1_c1% %bf1_c2% %bf1_c3% "
END
facility 8; event_type 13016;
aligned attributes {
	short	bf1_s1:3;
	char	:0;
	char	:0;
	short	bf1_s2:3;
	short	bf1_s3:3;
}
format string "natts = 3 \t%bf1_s1% %bf1_s2% %bf1_s3% "
END
facility 8; event_type 13017;
aligned attributes {
	short	bf1_s1:3;
	short	:0;
	char	:0;
	short	bf1_s2:3;
	short	bf1_s3:3;
}
format string "natts = 3 \t%bf1_s1% %bf1_s2% %bf1_s3% "
END
facility 8; event_type 13018;
aligned attributes {
	short	bf1_s1:3;
	char	:0;
	short	:0;
	short	bf1_s2:3;
	short	bf1_s3:3;
}
format string "natts = 3 \t%bf1_s1% %bf1_s2% %bf1_s3% "
END
facility 8; event_type 13019;
aligned attributes {
	int	bf1_i:3;
	char	:0;
	int	bf1_i2:3;
	short	:0;
	int	bf1_i3:3;
}
format string "natts = 3 \t%bf1_i% %bf1_i2% %bf1_i3% "
END
facility 8; event_type 13020;
aligned attributes {
	char	bf1_c:8;
	char	:0;
	short	bf1_s2:3;
	short	bf1_s3:3;
}
format string "natts = 3 \t%bf1_c% %bf1_s2% %bf1_s3% "
END
facility 8; event_type 13021;
aligned attributes {
	short bf1_s1:4;
	char bf1_c1:4;
	char bf1_c2:4;
	char bf1_c3:4;
	char bf1_c4:4;
	char c1;
}
format string "natts = 6 \t%bf1_s1% %bf1_c1% %bf1_c2% %bf1_c3% %bf1_c4% %c1% "
END
facility 8; event_type 13121;
aligned attributes {
	short bf1_s1:4;
	char bf1_c1:4;
	char bf1_c2:4;
	char bf1_c3:4;
	char bf1_c4:4;
	char c1:8;
}
format string "natts = 6 \t%bf1_s1% %bf1_c1% %bf1_c2% %bf1_c3% %bf1_c4% %c1% "
END
facility 8; event_type 13022;
aligned attributes {
	int	bf1_i:16;
	short	bf1_s:7;
	short	bf1_c:9;
}
format string "natts = 3 \t%bf1_i% %bf1_s% %bf1_c% "
END
facility 8; event_type 13023;
aligned attributes {
	char	bf1_c:7;
	short	bf1_s:10;
	int	bf1_i:23;
}
format string "natts = 3 \t%bf1_c% %bf1_s% %bf1_i% "
END
facility 8; event_type 13024;
aligned attributes {
	char	bf1_c:7;
	short	bf1_s:10;
	int	bf1_i:23;
	char	c;
	char	bf1_c2:7;
	short	bf1_s2:10;
	int	bf1_i2:23;
}
format string "natts = 7 \t%bf1_c% %bf1_s% %bf1_i% %c% %bf1_c2% %bf1_s2% %bf1_i2% "
END
facility 8; event_type 13124;
aligned attributes {
	char	bf1_c:7;
	short	bf1_s:10;
	int	bf1_i:23;
	char	c:8;
	char	bf1_c2:7;
	short	bf1_s2:10;
	int	bf1_i2:23;
}
format string "natts = 7 \t%bf1_c% %bf1_s% %bf1_i% %c% %bf1_c2% %bf1_s2% %bf1_i2% "
END
facility 8; event_type 13025;
aligned attributes {
	int	bf1_i:15;
	char	c1;
	char	c2;
}
format string "natts = 3 \t%bf1_i% %c1% %c2% "
END
facility 8; event_type 13125;
aligned attributes {
	int	bf1_i:15;
	char	c1:8;
	char	c2:8;
}
format string "natts = 3 \t%bf1_i% %c1% %c2% "
END
facility 8; event_type 13026;
aligned attributes {
	int	bf1_i:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c;
}
format string "natts = 4 \t%bf1_i% %bf1_s% %bf1_c% %c% "
END
facility 8; event_type 13126;
aligned attributes {
	int	bf1_i:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c:8;
}
format string "natts = 4 \t%bf1_i% %bf1_s% %bf1_c% %c% "
END
facility 8; event_type 13027;
aligned attributes {
	longlong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c;
}
format string "natts = 4 \t%bf1_ll% %bf1_s% %bf1_c% %c% "
END
facility 8; event_type 13127;
aligned attributes {
	longlong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c:8;
}
format string "natts = 4 \t%bf1_ll% %bf1_s% %bf1_c% %c% "
END
facility 8; event_type 13028;
aligned attributes {
	longlong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c;
	short	s1;
}
format string "natts = 5 \t%bf1_ll% %bf1_s% %bf1_c% %c% %s1% "
END
facility 8; event_type 13128;
aligned attributes {
	longlong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c:8;
	short	s1:16;
}
format string "natts = 5 \t%bf1_ll% %bf1_s% %bf1_c% %c% %s1% "
END
facility 8; event_type 13029;
aligned attributes {
	ulonglong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c;
	short	s1;
}
format string "natts = 5 \t%bf1_ll% %bf1_s% %bf1_c% %c% %s1% "
END
facility 8; event_type 13129;
aligned attributes {
	ulonglong	bf1_ll:2;
	short	bf1_s:2;
	char	bf1_c:2;
	char	c:8;
	short	s1:16;
}
format string "natts = 5 \t%bf1_ll% %bf1_s% %bf1_c% %c% %s1% "
END
facility 8; event_type 13030;
aligned attributes {
	longlong	bf1_ll:2;
	char	c;
	short	s1;
	longlong	ll1;
	int	i;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13130;
aligned attributes {
	longlong	bf1_ll:2;
	char	c:8;
	short	s1:16;
	longlong	ll1:64;
	int	i:32;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13031;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	longlong	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13032;
aligned attributes {
	xint	bf1_i:2;
}
format string "natts = 1 \t%bf1_i% "
END
facility 8; event_type 13033;
aligned attributes {
	int :2;
}
format string "natts = 0 \t"
END
facility 8; event_type 13035;
aligned attributes {
	char bf1_c1:2;
	int :2;
	char bf1_c2:2;
}
format string "natts = 2 \t%bf1_c1% %bf1_c2% "
END
facility 8; event_type 13036;
aligned attributes {
	longlong bf1_ll:40;
	short bf1_s:2;
}
format string "natts = 2 \t%bf1_ll% %bf1_s% "
END
facility 8; event_type 13037;
aligned attributes {
	longlong bf1_ll:40;
	short s;
}
format string "natts = 2 \t%bf1_ll% %s% "
END
facility 8; event_type 13137;
aligned attributes {
	longlong bf1_ll:40;
	short s:16;
}
format string "natts = 2 \t%bf1_ll% %s% "
END
facility 8; event_type 13038;
aligned attributes {
	char c;
	char bf1_c1:3;
	short bf1_s1:3;
	int bf1_i1:3;
	short s;
}
format string "natts = 5 \t%c% %bf1_c1% %bf1_s1% %bf1_i1% %s% "
END
facility 8; event_type 13138;
aligned attributes {
	char c:8;
	char bf1_c1:3;
	short bf1_s1:3;
	int bf1_i1:3;
	short s:16;
}
format string "natts = 5 \t%c% %bf1_c1% %bf1_s1% %bf1_i1% %s% "
END
facility 8; event_type 13039;
aligned attributes {
	char c;
	int bf1_i1:3;
	short bf1_s1:3;
	char bf1_c1:3;
	short s;
}
format string "natts = 5 \t%c% %bf1_i1% %bf1_s1% %bf1_c1% %s% "
END
facility 8; event_type 13139;
aligned attributes {
	char c:8;
	int bf1_i1:3;
	short bf1_s1:3;
	char bf1_c1:3;
	short s:16;
}
format string "natts = 5 \t%c% %bf1_i1% %bf1_s1% %bf1_c1% %s% "
END
facility 8; event_type 13040;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	char	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
	char	c;
}
format string "natts = 6 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% %c% "
END
facility 8; event_type 13140;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	char	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:3;
	char	c:8;
}
format string "natts = 6 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% %c% "
END
facility 8; event_type 13041;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	int	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:2;
	char	c;
}
format string "natts = 6 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% %c% "
END
facility 8; event_type 13141;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	int	:0;
	short	bf1_s3:3;
	short	bf1_s4:3;
	short	bf1_s5:2;
	char	c:8;
}
format string "natts = 6 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% %c% "
END
facility 8; event_type 13042;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	char	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13043;
aligned attributes {
	int	bf1_s1:3;
	int	bf1_s2:3;
	char	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13044;
aligned attributes {
	int	bf1_s1:3;
	int	bf1_s2:3;
	longlong	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13045;
aligned attributes {
	char	c;
	int	bf1_s1:3;
	int	bf1_s2:3;
	longlong	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 6 \t%c% %bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13145;
aligned attributes {
	char	c:8;
	int	bf1_s1:3;
	int	bf1_s2:3;
	longlong	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 6 \t%c% %bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13046;
aligned attributes {
	int	i;
	int	bf1_s1:3;
	int	bf1_s2:3;
	longlong	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 6 \t%i% %bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13146;
aligned attributes {
	int	i:32;
	int	bf1_s1:3;
	int	bf1_s2:3;
	longlong	:0;
	int	bf1_s3:3;
	int	bf1_s4:3;
	int	bf1_s5:3;
}
format string "natts = 6 \t%i% %bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13047;
aligned attributes {
	char	bf1_c:7;
	short	bf1_s:9;
	int	bf1_i:23;
}
format string "natts = 3 \t%bf1_c% %bf1_s% %bf1_i% "
END
facility 8; event_type 13052;
aligned attributes {
	short	bf1_s1:3;
	short	bf1_s2:3;
	short	bf1_s3:3;
	char	:0;
	short	bf1_s4:3;
	short	bf1_s5:3;
}
format string "natts = 5 \t%bf1_s1% %bf1_s2% %bf1_s3% %bf1_s4% %bf1_s5% "
END
facility 8; event_type 13053;
aligned attributes {
	char	c;
	char	bf1_c1:4;
	short	:0;
	int	bf1_i1:4;
}
format string "natts = 3 \t%c% %bf1_c1% %bf1_i1% "
END
facility 8; event_type 13153;
aligned attributes {
	char	c:8;
	char	bf1_c1:4;
	short	:0;
	int	bf1_i1:4;
}
format string "natts = 3 \t%c% %bf1_c1% %bf1_i1% "
END
facility 8; event_type 13054;
aligned attributes {
	char	c[3]ARRAY_FMT;
	int	bf1_i1:4;
}
format string "natts = 2 \t%c% %bf1_i1% "
END
facility 8; event_type 13055;
aligned attributes {
	char	c[3]ARRAY_FMT;
	int	bf1_i1:10;
}
format string "natts = 2 \t%c% %bf1_i1% "
END
facility 8; event_type 13056;
aligned attributes {
	struct b1	b1;
	int	bf1_i1:10;
}
format string "natts = 2 \t%b1% %bf1_i1% "
END
facility 8; event_type 13057;
aligned attributes {
	struct b1	b1[3]STRARRAY_FMT;
	int	bf1_i1:4;
}
format string "natts = 2 \t%b1% %bf1_i1% "
END
facility 8; event_type 13058;
aligned attributes {
	struct b2	b1[3]STRARRAY_FMT;
	int	bf1_i1:4;
}
format string "natts = 2 \t%b1% %bf1_i1% "
END
facility 8; event_type 13059;
aligned attributes {
	longlong bf1_ll:40;
	int bf1_i:2;
}
format string "natts = 2 \t%bf1_ll% %bf1_i% "
END
facility 8; event_type 13060;
aligned attributes {
	longlong bf1_ll:40;
	int bf1_i:8;
}
format string "natts = 2 \t%bf1_ll% %bf1_i% "
END
facility 8; event_type 13061;
aligned attributes {
	longlong bf1_ll:40;
	short x:15;
}
format string "natts = 2 \t%bf1_ll% %x% "
END
facility 8; event_type 13062;
aligned attributes {
	longlong bf1_ll:32;
	int :8;
	short bf1_s:2;
}
format string "natts = 2 \t%bf1_ll% %bf1_s% "
END
facility 8; event_type 13063;
aligned attributes {
	longlong bf1_ll:32;
	int :8;
	short s;
}
format string "natts = 2 \t%bf1_ll% %s% "
END
facility 8; event_type 13163;
aligned attributes {
	longlong bf1_ll:32;
	int :8;
	short s:16;
}
format string "natts = 2 \t%bf1_ll% %s% "
END
facility 8; event_type 13064;
aligned attributes {
	longlong bf1_ll:32;
	int :8;
	int bf1_i:2;
}
format string "natts = 2 \t%bf1_ll% %bf1_i% "
END
facility 8; event_type 13065;
aligned attributes {
	longlong bf1_ll:32;
	int :8;
	short x:15;
}
format string "natts = 2 \t%bf1_ll% %x% "
END
facility 8; event_type 13066;
aligned attributes {
	longlong bf1_ll:40;
	int bf1_i:25;
}
format string "natts = 2 \t%bf1_ll% %bf1_i% "
END
facility 8; event_type 13067;
aligned attributes {
	longlong	bf1_ll:2;
	char	c:8;
	short	s1:16;
	longlong	ll1:2;
	int	i:32;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13068;
aligned attributes {
	longlong	bf1_ll:2;
	char	c:8;
	short	s1:16;
	longlong	ll1:33;
	int	i:32;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13069;
aligned attributes {
	char	c:5;
	short	s:9;
	int	i:17;
}
format string "natts = 3 \t%c% %s% %i% "
END
facility 8; event_type 13070;
aligned attributes {
	longlong	bf1_ll:2;
	char	c:8;
	short	s1:16;
	longlong	ll1:33;
	int	i:7;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13071;
aligned attributes {
	int		i:2;
	longlong	ll:33;
}
format string "natts = 2 \t%i% %ll% "
END
facility 8; event_type 13072;
aligned attributes {
	longlong	bf1_ll:2;
	char	c:6;
	short	s1:8;
	longlong	ll1:33;
	int	i:10;
}
format string "natts = 5 \t%bf1_ll% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13073;
aligned attributes {
	int	i1:2;
	char	c:8;
	short	s1:16;
	longlong	ll1:33;
	int	i:32;
}
format string "natts = 5 \t%i1% %c% %s1% %ll1% %i% "
END
facility 8; event_type 13074;
aligned attributes {
	char	c1;
	char	c2;
	char	c3;
	longlong	ll1:33;
}
format string "natts = 4 \t%c1% %c2% %c3% %ll1% "
END
facility 8; event_type 13075;
aligned attributes {
	char	c1;
	char	c2;
	char	c3;
	char	c4;
	longlong	ll1:33;
}
format string "natts = 5 \t%c1% %c2% %c3% %c4% %ll1% "
END
facility 8; event_type 13076;
aligned attributes {
	char	c1;
	char	c2;
	char	c3;
	char	c4;
	char	c5;
	longlong	ll1:32;
}
format string "natts = 6 \t%c1% %c2% %c3% %c4% %c5% %ll1% "
END
