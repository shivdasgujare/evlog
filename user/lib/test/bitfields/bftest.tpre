#define ARRAY_FMT "(%d,)"
#define STRARRAY_FMT "(%Z,)"
typedef int xint;
typedef uint uxint;

struct b1;
aligned attributes {
	char c;
}
format string "{%c%}"
END

struct b2;
aligned attributes {
	char c1:4;
	char c2:4;
}
format string "{%c1%,%c2%}"
END
