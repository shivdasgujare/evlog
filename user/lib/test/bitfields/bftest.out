3001: natts = 3 	1 1 1 
3002: natts = 3 	1 1 1 
3003: natts = 2 	1 1 
3004: natts = 5 	1 1 1 1 1 
3005: natts = 5 	1 1 1 1 1 
3006: natts = 4 	1 1 1 1 
3007: natts = 4 	1 1 1 1 
3008: natts = 3 	1 1 1 
3010: natts = 5 	1 1 1 1 1 
3011: natts = 5 	1 1 1 1 1 
3012: natts = 5 	1 1 1 1 1 
3013: natts = 5 	1 1 1 1 1 
3014: natts = 3 	1 1 1 
3015: natts = 3 	1 1 1 
3016: natts = 3 	1 1 1 
3017: natts = 3 	1 1 1 
3018: natts = 3 	1 1 1 
3019: natts = 3 	1 1 1 
3020: natts = 3 	1 1 1 
3021: natts = 6 	1 1 1 1 1 1 
3121: natts = 6 	1 1 1 1 1 1 
3022: natts = 3 	1 1 1 
3023: natts = 3 	1 1 1 
3024: natts = 7 	1 1 1 1 1 1 1 
3124: natts = 7 	1 1 1 1 1 1 1 
3025: natts = 3 	1 1 1 
3125: natts = 3 	1 1 1 
3026: natts = 4 	1 1 1 1 
3126: natts = 4 	1 1 1 1 
3027: natts = 4 	1 1 1 1 
3127: natts = 4 	1 1 1 1 
3028: natts = 5 	1 1 1 1 1 
3128: natts = 5 	1 1 1 1 1 
3029: natts = 5 	1 1 1 1 1 
3129: natts = 5 	1 1 1 1 1 
3030: natts = 5 	1 1 1 1 1 
3130: natts = 5 	1 1 1 1 1 
3031: natts = 5 	1 1 1 1 1 
3032: natts = 1 	1 
3033: natts = 0 	
3035: natts = 2 	1 1 
3036: natts = 2 	1 1 
3037: natts = 2 	1 1 
3137: natts = 2 	1 1 
3038: natts = 5 	1 1 1 1 1 
3138: natts = 5 	1 1 1 1 1 
3039: natts = 5 	1 1 1 1 1 
3139: natts = 5 	1 1 1 1 1 
3040: natts = 6 	1 1 1 1 1 1 
3140: natts = 6 	1 1 1 1 1 1 
3041: natts = 6 	1 1 1 1 1 1 
3141: natts = 6 	1 1 1 1 1 1 
3042: natts = 5 	1 1 1 1 1 
3043: natts = 5 	1 1 1 1 1 
3044: natts = 5 	1 1 1 1 1 
3045: natts = 6 	1 1 1 1 1 1 
3145: natts = 6 	1 1 1 1 1 1 
3046: natts = 6 	1 1 1 1 1 1 
3146: natts = 6 	1 1 1 1 1 1 
3047: natts = 3 	1 1 1 
3052: natts = 5 	1 1 1 1 1 
3053: natts = 3 	1 1 1 
3153: natts = 3 	1 1 1 
3054: natts = 2 	1,1,1, 1 
3055: natts = 2 	1,1,1, 1 
3056: natts = 2 	{1} 1 
3057: natts = 2 	{1},{1},{1}, 1 
3058: natts = 2 	{1,1},{1,1},{1,1}, 1 
3059: natts = 2 	1 1 
3060: natts = 2 	1 1 
3061: natts = 2 	1 1 
3062: natts = 2 	1 1 
3063: natts = 2 	1 1 
3163: natts = 2 	1 1 
3064: natts = 2 	1 1 
3065: natts = 2 	1 1 
3066: natts = 2 	1 1 
3067: natts = 5 	1 1 1 1 1 
3068: natts = 5 	1 1 1 1 1 
3069: natts = 3 	1 1 1 
3070: natts = 5 	1 1 1 1 1 
3071: natts = 2 	1 1 
3072: natts = 5 	1 1 1 1 1 
3073: natts = 5 	1 1 1 1 1 
3074: natts = 4 	1 1 1 1 
3075: natts = 5 	1 1 1 1 1 
3076: natts = 6 	1 1 1 1 1 1 
